package/MDAnalysis/topology/LAMMPSParser.py

```diff
@@ -256,8 +256,12 @@ def _parse_pos(self, datalines, pos):
         """Strip coordinate info into np array"""
         for line in datalines:
             line = line.split()
+            n = len(line)
             idx = int(line[0]) - 1
-            x, y, z = map(float, line[-3:])
+            if n in (7, 10):
+                x, y, z = map(float, line[4:7])
+            elif n in (6, 9):
+                x, y, z = map(float, line[3:6])
```

Comment:
Also `np.float32` accepts list so no need for map. And another thing please prefer list comprehension to map. It does the same but people are more likely to have seem the comprehension